idb/idb.go

```diff
@@ -240,12 +240,12 @@ type TransactionFilter struct {
 
 	Limit uint64
 
-	// If this flag is set to true, then the query returns the inner txn
-	// instead of the root txn.
-	ReturnInnerTxnOnly bool
+	// If this flag is set to true, then the query returns inner transactions
+	// instead of resolving them as the root transaction.
+	IncludeInnerTxns bool
 
-	// If this flag is set to true, then the query returns only root txns
-	// and no inner txns
+	// If this flag is set to true, then the query only returns root
+	// transactions and skips matching or returning inner transactions.
```

Comment:
This is confusing me--the descriptions sound like they do the same thing. That being, return only root txns instead of inner txns. Or in other words, `!IncludeInnerTxns == ReturnRootTxnsOnly`. But I think they're actually different.

Reply:
Aah back to the drawing board. When you say it that way these names still sound pretty bad.  I agree though, the flags are confusing. I think it's because what we're doing isn't intuitive, so there is no basis for what the variables are doing.

The unintuitive behavior: when the query matches an inner transaction the root transaction is included in the result and the API returns only the root transaction.

The fields, and obnoxious but maybe more technically correct names...

* IncludeInnerTxns -> DoNotConvertInnerTxnToRootTxn
  * Actually return the inner transaction instead of converting it into the root transaction.

* ReturnRootTxnsOnly -> SkipInnerTransactionsInQuery
  * Skip inner transactions at the query level, completely side-stepping the conversion process